### PR TITLE
chore: cleanup deprecated icon library references

### PR DESCRIPTION
**Pull request by smfonseca**
## Description:
This PR removes references to the deprecated `global-resources` icon library and updates some of the icons to their new counterpart.

## Definition of Reviewable:

- [ ] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated

- [ ] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked

### CODE REVIEW

**Comment by smfonseca** [comment]
@coraliefeil @MartaPintoTeixeira there are a few changes that need to be reviewed in chromatic.

File: .changeset/flat-eyes-guess.md
Comment (mariohamann) [inline .changeset/flat-eyes-guess.md:1]: Did you have a look in https://github.com/solid-design-system/solid/blob/main/packages/docs/.storybook/preview.js as well? Maybe you have to update some mocks there.

Comment (mariohamann) [reply .changeset/flat-eyes-guess.md:1]: Reason behind the mocks: In the past we had sometimes issues when there were to many requests to the CDN. I would be fine with getting rid of the icon mocks to try it out.

Comment (smfonseca) [reply .changeset/flat-eyes-guess.md:1]: Thanks for the backstory. I've removed the mocks so that we can test it

**Review by paulovareiro29 — DISMISSED** [review]
LGTM

**Review by auroraVasconcelos — APPROVED** [review]

**Comment by MartaPintoTeixeira** [comment]
As spoken while checking this ticket I have noticed a bug on sd-tab.
Opened up a ticket for it: Fix space between icon and text for sd-tab
https://github.com/solid-design-system/solid/issues/1709

**Review by MartaPintoTeixeira — APPROVED** [review]

**Review by paulovareiro29 — APPROVED** [review]

**Pull request merged**